### PR TITLE
Update medication terms

### PR DESCRIPTION
**Pull request by acrollet**
## Summary

This PR updates the medication terms widget (as shown in screenshot) to use dynamic data and be shown in another location on the AVS.

- Uses pharmacy terms from backend data (instead of hard-coded PoC)
- Adds new component to Medication Orders section
- Adds unit tests for component

## Testing done

- Manual testing + new unit test

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  | image | image  |

## What areas of the site does it impact?

- AVS application

## Acceptance criteria

- Medication Terms widget is shown under `Your appointment on <Date> -> Medications ordered for administration in clinic`
- Medication Terms widget is shown under `Your treatment plan from this appointment -> New orders from this appointment -> Medications & supplies`
- Medication Terms widget is not shown if `avs.pharmacyTerms` is empty

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### CODE REVIEW

File: src/applications/avs/components/MedicationTerms.jsx [hunk 1]
```diff
@@ -1,25 +1,31 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+
+const MedicationTerms = props => {
+  const { avs } = props;
+
+  if (!avs.pharmacyTerms?.length) {
+    return null;
```
Comment (carlosfelix2) [inline src/applications/avs/components/MedicationTerms.jsx:8]: Are you sure? Wouldn't a message saying something like "you don't have any medications" be better? I can't see the entire page from just the screenshots to know if there is anything else there.

File: src/applications/avs/components/MedicationTerms.jsx [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
-        supplement, or sample medication. If this medication information is
-        incorrect or out of date, please tell your VA healthcare team.
-      </p>
+      {avs.pharmacyTerms.map((term, idx) => (
+        <div key={`term-${idx}`}>
+          <h4>
+            {term.term} {!!term.aka && term.aka}
+          </h4>
+          <p>
+            {term.explanation} {!!term.patientActions && term.patientActions}
```
Comment (carlosfelix2) [inline src/applications/avs/components/MedicationTerms.jsx:19]: Screenshots don't show most of these fields. Would be nice to see how this shows up.

File: src/applications/avs/tests/components/MedicationTerms.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  it('sections without data are hidden', async () => {
+    const avsCopy = replacementFunctions.cloneDeep(avsData);
+    delete avsCopy.pharmacyTerms;
+    const props = {
+      avs: avsCopy,
+    };
+    const screen = render(<MedicationTerms {...props} />);
+    expect(screen.queryByRole('heading')).to.not.exist;
```
Comment (carlosfelix2) [inline src/applications/avs/tests/components/MedicationTerms.unit.spec.jsx:28]: How about testing the other fields are also shown?

**Review by carlosfelix2 — APPROVED** [review]

File: src/applications/avs/components/MedicationTerms.jsx (hunk 1 above)
Comment (acrollet) [reply src/applications/avs/components/MedicationTerms.jsx:8]: @kristenmcconnell @ninaranusavice  any thoughts on this? I'm assuming that we wouldn't show the widget at all if there are no terms to define, but that could have been a bad assumption 🙂

**Review by powellkerry — APPROVED** [review]

File: src/applications/avs/tests/components/MedicationTerms.unit.spec.jsx (hunk 3 above)
Comment (acrollet) [reply src/applications/avs/tests/components/MedicationTerms.unit.spec.jsx:28]: addressing in follow-on :)

**Pull request merged**